### ISSUE TITLE
[MNT]: Cleanup Timer inheritance scheme

### ISSUE THREAD

**Issue by anntzer** [matplotlib/matplotlib#26855]
### Summary

Currently, subclasses inheriting from TimerBase need to override a number of private methods (at least _timer_start() and _timer_stop(), as well as optionally a few others).  It would be better (e.g. for documentation purposes, as argued elsewhere) if only public methods needed to be overridden.

At a first glance this should be fairly doable, by directly overriding start() and stop() instead (possibly getting rid of the interval kwarg to start(), which we never use and can be replaced by `timer = Timer(interval=...)` or `timer.interval = ...` before starting anyways) and reading the values off timer.interval and timer.single_shot directly.  Of course, the change would need to go through deprecate_method_override, yada yada.

**Comment by timhoffm** [comment]
Sounds ok and doable. I think it's a low priority, so I wouldn't start it myself, but if you want to make a PR, go for it.

**Comment by SunSummoner** [comment]
Hi

I'd like to make a PR for this issue
